### PR TITLE
Fix #21 and prep for a release.

### LINKED ISSUE #21: When chef_handler reloads a file, any transitive "require"ments are not reloaded as well.
We just pulled down v 1.1.9 and immediately our chef-runs broke with the following error:

```  ================================================================================
  Recipe Compile Error in /var/chef/cache/cookbooks/box-sonar-01/recipes/default.rb
  ================================================================================
  
  NoMethodError
  -------------
  chef_handler[Chef::Handler::Datadog] (datadog::dd-handler line 30) had an error: NoMethodError: undefined method `new' for ChefHandler:Module
  
  Cookbook Trace:
  ---------------
    /var/chef/cache/cookbooks/datadog/recipes/dd-handler.rb:40:in `from_file'
    /var/chef/cache/cookbooks/linux-base/recipes/default.rb:28:in `from_file'
    /var/chef/cache/cookbooks/sonarqube-cvent/recipes/default.rb:20:in `from_file'
    /var/chef/cache/cookbooks/box-sonar-01/recipes/default.rb:15:in `from_file'
  
  Relevant File Content:
  ----------------------
  /var/chef/cache/cookbooks/datadog/recipes/dd-handler.rb:
  
   33:      :api_key => node['datadog']['api_key'],
   34:      :application_key => node['datadog']['application_key'],
   35:      :use_ec2_instance_id => node['datadog']['use_ec2_instance_id'],
   36:      :notify_on_failure => node['datadog']['handler']['notify_on_failure']
   37:    ]
   38:    supports :report => true, :exception => true
   39:    action :nothing
   40>> end.run_action(:enable) if node['datadog']['chef_handler_enable']
   41:  
```

As a workaround, reverting to 1.1.8 resolved the issue.

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Requirements
 ============
 
 * Ruby >= 1.9
+* Chef client >= 12
```
Comment (coderanger) [inline README.md]: This doesn't actually require Chef client, just `Chef >= 12`. Viva la solo!

**Comment by coderanger** [comment]
Can probably be even more minimal, but this is a good short-term bug fix :+1:

**Comment by ksubrama** [comment]
Yep - now that we don't need all that class loading stuff, we don't really need a "helper" file with those teeny-weeny routines.  But I was lazy and didn't feel like breaking all the things :-P

**Comment by coderanger** [comment]
Nuke it from orbit

**Comment by miketheman** [comment]
I saw this was released as a 1.2.0 release - with clearly defined breaking changed for people not on Chef 12 yet.

I'd consider that a major, backwards-breaking change, and would bump the cookbook to version 2.0.0, especially based on some good guidelines set forth in http://chef-community.github.io/cvp/

Thoughts?

**Comment by ksubrama** [comment]
If there was major functionality change, I'd agree.  In this case, there were issues with the cookbook's attempt to handle reloading handler files in long-running chef client runs and the breakage was present since 1.1.0 (i,e sometime in 2012).  It was a "no right answer for all scenarios" issue.  Anybody who actually cared about this has since worked around it.  I had attempted to fix this code in 1.1.19 and created a new class of failures (as shown in #21 ).  After being told that we no longer have long running processes and fork a new process for each chef run, I went in and just yanked all that code out.  I believe this was sometime during Chef 11 - I'm not sure exactly when but for all common use-cases, I believe this cookbook should still work in Chef-11.  Since this wasn't so much a "backwards-breaking" change as much as it was a "backwards-no-more-broken-than-before", I didn't bump the major version.  If you noticed no issues moving from 1.1.18 to 1.1.19, you won't (or rather shouldn't) notice any functional difference with this move to 1.2.0. You can still use it with Chef-11.  If I find the exact time when the forking change happens, I'll update the README to add that detail.

**Comment by miketheman** [comment]
@ksubrama thanks for the speedy response. I'll admit - I haven't tested the cookbook in this scenario - and we're still on Chef 11, which prompted my concern.

In README, you updated a hard requirement for Chef 12 and above, so if from the outcome of "running chef 11 as a daemon still works with this cookbook", then that's great news, and the requirements should be updated, like yo mentioned.

Looking forward to more improvements, and thanks for your hard work!

**Comment by coderanger** [comment]
This is absolutely a breaking change and should be a major version release.

**Pull request merged**